yrs/src/types/text.rs

```diff
@@ -128,8 +122,16 @@ impl Text {
     ///
     /// This method will panic if provided `index` is greater than the length of a current text.
     pub fn insert(&self, txn: &mut Transaction, index: u32, chunk: &str) {
-        if let Some(pos) = self.find_position(txn, index) {
+        if let Some(mut pos) = self.find_position(txn, index) {
             let value = crate::block::PrelimText(chunk.into());
+            while let Some(right) = pos.right.as_ref() {
+                if right.is_deleted() {
```

Comment:
Fix: after finding a left origin, we keep skipping over deleted blocks before inserting a new block.